File: commons/src/main/scala/org/dinky/udf/scala/SubstringFunction.scala
package org.dinky.udf.scala

import org.apache.flink.table.functions.ScalarFunction

class SubstringFunction extends ScalarFunction {
  def eval(s: String, begin: Integer, end: Integer): String = {
    println("this is scala")
    s.substring(begin, end)
  }
}

File: commons/src/main/java/org/dinky/udf/SubstringFunction.java
package org.dinky.udf;

import org.apache.flink.table.functions.ScalarFunction;

/**
 * @author ZackYoung
 * @since 1.0.0
 */
public class SubstringFunction extends ScalarFunction {
    public String eval(String s, Integer begin, Integer end) {
        System.out.println("this is java");
        return s.substring(begin, end);
    }
}
